internal/dag/accessors.go

```diff
@@ -103,13 +103,38 @@ func validateExternalName(svc *v1.Service, enableExternalNameSvc bool) error {
 	return nil
 }
 
+// the ServicePort's AppProtocol must be one of the these.
+const (
+	protoK8sH2C     = "kubernetes.io/h2c"
+	protoK8sWS      = "kubernetes.io/ws"
+	protoContourH2  = "projectcontour.io/h2"
+	protoContourTLS = "projectcontour.io/tls"
+)
+
+func toContourProtocol(appProtocol string) (string, bool) {
+	proto, ok := map[string]string{
+		// *NOTE: for gateway-api: the websocket is enabled by default
+		protoK8sWS:  "",
```

Comment:
Thinking about this some more, websocket upgrades are already enabled by default for Gateway API, and this PR does not enable using `kubernetes.io/ws` for HTTPProxy or Ingress, so I think we should _either_:
- not mention `kubernetes.io/ws` in the changelog, and instead just focus it on supporting the `h2c` value; _or_ 
- implement support for `kubernetes.io/ws` for Ingress and HTTPProxy in this PR to be consistent with the changelog

@izturn which would you like to do here?

Reply:
I think we can focus on passing the conformance test first, and then I can send a change that implements option 2 later.